admin/auth_token.go

```diff
@@ -84,3 +84,8 @@ func (s *Service) RevokeAuthToken(ctx context.Context, token string) error {
 
 	return s.DB.DeleteUserAuthToken(ctx, parsed.ID.String())
 }
+
+// RevokeAuthTokenByID removes an auth token from persistent storage.
+func (s *Service) RevokeAuthTokenByID(ctx context.Context, tokenID string) error {
+	return s.DB.DeleteUserAuthToken(ctx, tokenID)
+}
```

Comment:
If we add `service_auth_tokens`, this breaks. Probably remove this func, and just call `s.DB.DeleteUserAuthToken` directly – then it will have context about the tokenID actually being a user token

Reply:
removed and doing direct call. However isn't it doing similar things as `RevokeAuthToken` where `s.DB.DeleteUserAuthToken(ctx, parsed.ID.String())` is called. I created this method to prevent creating token string again before calling `RevokeAuthToken`

Reply:
You're right that's a bug in `RevokeAuthToken` – it should check the type of `parsed.Type` (like in `ValidateAuthToken`). Can you make that fix?

You can just rename the func here to `RevokeUserAuthTokenByID`, then it will be good. The place you're calling it, you have a check on the claims type, so you know it's a user token